jac-uk/admin
# #1728 Exercise Approval

## What's included?
As a user who can update exercises i can submit a draft exercise for approval.
Upon submission the exercise will move to the 'ready' state and an email will be sent to the Senior Selection Exercise Mgrs to notify them of this request.
The exercise can be approved or rejected (with a specified reason) and either change will be displayed on the view exercise page.
As a user who can approve exercises i can approve or reject an exercise which is in the 'ready' state. On reject i have to specify a reason. Once approved or rejected if i view the exercise i will be able to see that it has been approved or rejected with a reason.

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?

1)
a) Login as a user who can update exercises (but not approve them). e.g. jacstaff@judicialappointments.digital
b) Find an exercise in the 'draft' state and scroll to the bottom of the page and click the Submit For Approval button.
c) A modal will appear.
d) If you click the Cancel button the modal will disappear.
e) If you click the Submit For Approval button the modal will disappear and the Submit For Approval button (near the bottom of the page) will also disappear. A warning message displays near the top informing the user that 'This exercise is being checked by SLs...'. An email should be sent to the SLs.
f) If the exercise is subsequently approved then this page will no longer display the warning message.
g) If the exercise is subsequently rejected then this page will display a warning message saying 'This exercise was not approved by the SLs...' and also it will detail the rejection reason underneath the warning. The Submit For Approval button will reappear allowing the user to resubmit the exercise for approval.

2)
a) Login as a user who can approve exercises (but not update them). e.g.  seniorleader@judicialappointments.digital
b) Find an exercise in the 'ready' state. It should have a warning message near the top saying 'This exercise is waiting for your approval...' with  Reject and Approve button beneath.
c) If the Reject button is clicked then the warning message changes to read 'You are rejecting an exercise...' and a text box appears asking you to add a rejection reason with a Reject and Cancel button beneath it. If the Cancel button is clicked then the previous message and buttons appear as in 2a above. If the Reject button is clicked then a warning message appears saying 'This exercise was rejected by you' with the rejection reason below.
d) If the Approve button is clicked then a message appears saying 'The exercise was approved by you'.

3)
a) Login as a user who can BOTH update exercises AND approve them. e.g.  seniorleader@judicialappointments.digital
b) Find an exercise in the 'draft' state and scroll to the bottom of the page and click the Submit For Approval button.
c) A modal will appear.
d) If you click the Cancel button the modal will disappear.
e) If you click the Submit For Approval button the modal will disappear and the Submit For Approval button (near the bottom of the page) will also disappear. A warning message will appear near the top saying 'This exercise is waiting for your approval...' with  Reject and Approve button beneath.
f) If the Reject button is clicked then the warning message changes to read 'You are rejecting an exercise...' and a text box appears asking you to add a rejection reason with a Reject and Cancel button beneath it. If the Cancel button is clicked then the previous message and buttons appear as in 2a above. If the Reject button is clicked then a warning message appears saying 'This exercise was rejected by you' with the rejection reason below.
g) If the Approve button is clicked then a message appears saying 'The exercise was approved by you'.

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
